### PR TITLE
[Imagemagick 6 / 7] Correct HDRI troubles with gemacs

### LINKED ISSUE #3702: [gemacs] Does not display since imagemagick 6.9.10-77
While running gemacs, following error is displayed :
```
gemacs: error while loading shared libraries: libMagickWand-6.Q16.so.6: cannot open shared object file: No such file or directory
```
Trying to remove / reinstall from sources gemacs or imagemagick doesn't help.

### PR DESCRIPTION
**Pull request by vincowl**
Fixes #3702

### CODE REVIEW

File: packages/imagemagick6.rb [hunk 1]
```diff
@@ -64,5 +64,12 @@ def self.build
 
   def self.install
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    system "ln -s #{CREW_LIB_PREFIX}/libMagickWand-6.Q16HDRI.so.6 #{CREW_LIB_PREFIX}/libMagickWand-6.Q16.so.6"
+    system "ln -s #{CREW_LIB_PREFIX}/libMagickCore-6.Q16HDRI.so.6 #{CREW_LIB_PREFIX}/libMagickCore-6.Q16.so.6"
```
Comment (uberhacker) [inline packages/imagemagick6.rb]: I believe you will need:
```
FileUtils.ln_s "#{CREW_LIB_PREFIX}/libMagickWand-6.Q16HDRI.so.6", "#{CREW_DEST_LIB_PREFIX}/libMagickWand-6.Q16.so.6"
FileUtils.ln_s "#{CREW_LIB_PREFIX}/libMagickCore-6.Q16HDRI.so.6", "#{CREW_DEST_LIB_PREFIX}/libMagickCore-6.Q16.so.6"
```

File: packages/imagemagick6.rb [hunk 2]
```diff
@@ -64,5 +64,12 @@ def self.build
...
+  end
+  
+  def self.remove
```
Comment (uberhacker) [inline packages/imagemagick6.rb]: I don't believe we have a self.remove method.  Generally, you won't need to remove files if the files have been tracked.  If you apply the suggestion above, you won't need to do this.

File: packages/imagemagick7.rb [hunk 3]
```diff
@@ -64,5 +64,12 @@ def self.build
 
   def self.install
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    system "ln -s #{CREW_LIB_PREFIX}/libMagickWand-7.Q16HDRI.so.6 #{CREW_LIB_PREFIX}/libMagickWand-7.Q16.so.6"
+    system "ln -s #{CREW_LIB_PREFIX}/libMagickCore-7.Q16HDRI.so.6 #{CREW_LIB_PREFIX}/libMagickCore-7.Q16.so.6"
```
Comment (uberhacker) [inline packages/imagemagick7.rb]: See comment in imagemagick6.rb.

File: packages/imagemagick7.rb [hunk 4]
```diff
@@ -64,5 +64,12 @@ def self.build
...
+  end
+
+  def self.remove
```
Comment (uberhacker) [inline packages/imagemagick7.rb]: See comment in imagemagick6.rb.

File: packages/imagemagick7.rb [hunk 5]
```diff
@@ -64,5 +64,12 @@ def self.build
...
+    system "rm #{CREW_LIB_PREFIX}/libMagickWand-7.Q16.so.6"
+    system "rm #{CREW_LIB_PREFIX}/libMagickCore-7.Q16.so.6"
```
Comment (uberhacker) [inline packages/imagemagick7.rb]: We will also need new binaries so please remove the existing `binary_url` and `binary_sha256` sections.

**Comment by uberhacker** [comment]
Thank you for the update @vincowl.  Please see comments.

File: packages/imagemagick6.rb (hunk 2 above)
Comment (vincowl) [reply packages/imagemagick6.rb]: Could you briefly explain how files are tracked ? I don't really understand this point and that will help me to be more productive next time.

File: packages/imagemagick7.rb (hunk 5 above)
Comment (vincowl) [reply packages/imagemagick7.rb]: I will apply it to `imagemagick6.rb` too

File: packages/imagemagick6.rb (hunk 2 above)
Comment (uberhacker) [reply packages/imagemagick6.rb]: When you add files to the `CREW_BREW_DIR` and subdirectories, these files are tracked for installation and therefore are available for removal as well.  If you get a chance, study the `crew` command to get a better idea of how this works.

Comment (vincowl) [reply packages/imagemagick6.rb]: Thanks !

**Pull request merged**